src/integrations/stripe.ts

```diff
@@ -127,6 +139,51 @@ export async function setupPaymentMethod(
           type: servar.type,
           displayText: servar.name
         };
+
+        //
+        // Payment sequence
+        //
+        const { intent_secret: paymentIntentSecret } =
+          await client.createPayment(servar.key);
+        // No paymentIntentSecret means no payment will be done
+        if (paymentIntentSecret) {
+          const result = await stripe.confirmCardPayment(paymentIntentSecret, {
+            payment_method: paymentMethodId
+          });
+          if (result.error)
+            return {
+              errorField: servar,
+              errorMessage: result.error.message
+            };
+          else {
+            // Complete the payment (clears fields, sets paid indicator)
+            const { product_fields_to_clear: productFieldKeys } =
+              await client.paymentComplete();
+
+            // clear the selected product hidden fields
+            productFieldKeys.forEach((key: string) =>
+              updateFieldValues({
+                [key]: ''
+              })
+            );
+            // clear the total in the total hidden field if mapped (leaving the currency symbol off)
+            const totalFieldKey =
+              integrationData.metadata?.payment_field_mappings?.total;
+            if (totalFieldKey) {
+              updateFieldValues({
+                [totalFieldKey]: '0.00'
```

Comment:
shoudl this be a float instead of string?

Reply:
This gets presented on the ui using a text var.  AFAIK, there is no formatting supported with text var, so need to format it nicely in the hidden field as a monetary format.  Not ideal, I agree.  Let me know if there is another way.

Reply:
All of this logic is moving to the BE so not ex-licit change is needed here.  However, there is a reason to format this as a string.  The designer will be displaying this total using a text var which has not formatting functionality.  This needs to display as a monetary amount.